sdk/core/Azure.Core/src/Pipeline/RetryPolicy.Adapter.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Core.Pipeline;
+
+public partial class RetryPolicy
+{
+    /// <summary>
+    /// Adds Azure.Core features to the System.ClientModel retry policy.
+    /// </summary>
+    internal class AzureCoreRetryPolicy : ClientRetryPolicy
+    {
+        private readonly RetryPolicy _pipelinePolicy;
+        private readonly DelayStrategy _delayStrategy;
+
+        private long _beforeProcess;
+        private long _afterProcess;
+        private double _elapsedTime;
```

Comment:
Is this instance of `AzureCoreRetryPolicy` constructed per request? If not, the scope of these state variables seems unsafe.

Reply:
Ah, excellent catch, and thank you.  It is not, and I don't think we would want to make that allocation per call to pipeline.Send.  Let me look at this again -- worst case we can stash those values in the message property bag, but I'm interested to see if there is another approach to solve this.  Thanks!

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-net/pull/42329